### PR TITLE
[VL] [PoC] Velox backend: Package uber jar with shaded protobuf

### PR DESCRIPTION
**Pull request by zhztheplayer**
#789

### CODE REVIEW

File: package/velox/spark32/pom.xml [hunk 1]
```diff
@@ -40,17 +40,22 @@
...
             <phase>package</phase>
             <goals>
-              <goal>single</goal>
+              <goal>shade</goal>
             </goals>
+            <configuration>
+              <relocations>
+                <relocation>
+                  <pattern>com.google.protobuf</pattern>
+                  <shadedPattern>shaded.com.google.protobuf</shadedPattern>
```
Comment (LuciferYang) [inline package/velox/spark32/pom.xml:55]: I suggest use a special `shadedPattern` prefix, such as `io-glutenproject`, `shaded` is too generic and easy to conflict with other projects

Comment (zhztheplayer) [reply package/velox/spark32/pom.xml:55]: Fair enough to me. If the PoC can be proved feasible then we'll use a special pattern than `shaded` in the final patch.

**Comment by FelixYBW** [comment]
@zhztheplayer should we keep it open?

**Comment by CodingCat** [comment]
I will back to this soon, it needs a bit tweak based on latest master branch but generally working

**Comment by FelixYBW** [comment]
> I will back to this soon, it needs a bit tweak based on latest master branch but generally working

Thank you!

**Comment by zhztheplayer** [comment]
Thanks @CodingCat . Feel free to raise another patch to cover this one.

**Comment by zhztheplayer** [comment]
Closed by https://github.com/oap-project/gluten/pull/1030

**Pull request closed without merging**